draft-wang-ppm-differential-privacy.md

```diff
@@ -322,28 +408,42 @@ with the correct noise type.
 
 > TODO: Chris P: we will mention Prio3SumVec because that's what we use to describe aggregator DP with amplification
 
+# Primitives for Implementing DP Mechanisms {#primitives}
 
-# Differential Privacy Mechanism {#dp mechanism}
+This section describes various primitives required for implementing DP
+mechansims. The algorithms are designed to securely expand a short, uniform
+random seed into a sample from a given distribution.
 
-This section should define a framework to apply different dp noise type to a
-client measurement and/or an aggregation.
+## Discrete Laplace
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
+> TODO: Specify a Laplace sampler
```

Comment:
I think so, but we could probably reuse the PRGs defined in the VDAF draft.

The purpose is to have a reference implementation of these algorithms so that we increase the likelihood of others developing secure implementations. (My understanding is that it's easy to mess up here in a way that invalidates DP.) If the algorithm spells out expanding a seed into a sample from the given distribution, then we can produce test vectors to help increase confidence.

Could you file an issue for specifying algorithms {{CKS20}}? No reason to actually do the work between now and Monday, but it'll be good to have our work scoped out.

@tholop may be able to help out here.